### PR TITLE
Trial PR for using scalafmt

### PR DESCRIPTION
**Pull request by chick**
- enables sbt plugin for scalafmt in sbt
- adds travis test requiring scalafmt for PR
- formats everything in src/
- formats build.sbt

### Contributor Checklist

- [x] Did you add Scaladoc to every public function/method?
- [NA] Did you update the FIRRTL spec to include every new feature/behavior?
- [x] Did you add at least one test demonstrating the PR?
- [NA] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [x] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

Code refactor. Style only. Force code to follows standard scalafmt

#### API Impact

No impact on API

#### Backend Code Generation Impact

No consequences for code generation

#### Desired Merge Strategy

Squash on merge

#### Release Notes

From this PR on all firrtl PR must follow scalafmt rules. This will be enforced in CI.

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (1.2.x, 1.3.0, 1.4.0) ?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you mark as `Please Merge`?

### CODE REVIEW

**Comment by albertchen-sifive** [comment]
LGTM, but requesting review from @freechipsproject/firrtl-reviewers since I don't really know what the agreed upon `.scalafmt.conf` is supposed to look like. can you push a commit with a bad formatting change to one of the .scala files just to sanity check CI.

File: src/main/scala/firrtl/Compiler.scala [hunk 1]
```diff
@@ -324,8 +334,9 @@ trait Transform extends TransformLike[CircuitState] with DependencyAPI[Transform
       Dependency[SystemVerilogEmitter] :: Nil
 
     val emitters = inputForm match {
-      case C => Dependency[ChirrtlEmitter]      :: Dependency[HighFirrtlEmitter]   :: Dependency[MiddleFirrtlEmitter] :: lowEmitters
-      case H => Dependency[HighFirrtlEmitter]   :: Dependency[MiddleFirrtlEmitter] :: lowEmitters
+      case C =>
+        Dependency[ChirrtlEmitter] :: Dependency[HighFirrtlEmitter] :: Dependency[MiddleFirrtlEmitter] :: lowEmitters
```
Comment (sequencer) [inline src/main/scala/firrtl/Compiler.scala:338]: I think words per line limitation can be increased? Maybe 120 or 180?

Comment (jackkoenig) [reply src/main/scala/firrtl/Compiler.scala:338]: It's currently set to 120, I don't think we should go higher than that.

**Comment by chick** [comment]
Superseded by PR #1848

**Pull request closed without merging**